gcloud/pubsub/client.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcloud._helpers import get_default_project
+from gcloud._helpers import _ClientProxy
+from gcloud.pubsub._implicit_environ import _require_connection
+from gcloud.pubsub import api
+from gcloud.pubsub.subscription import Subscription
+from gcloud.pubsub.topic import Topic
+
+
+class Client(object):
+    """Wrap :mod:`gcloud.pubsub` API objects.
+
+    :type connection: :class:`gcloud.pubsub.connection.Connection` or None
+    :param connection: The configured connection. Defaults to one inferred
+                       from the environment.
+
+    :type project: str or None
+    :param connection: The configured project. Defaults to the value inferred
+                       from the environment.
+    """
+
+    def __init__(self, connection=None, project=None):
```

Comment:
Wait -- doesn't that just seem like extra useless typing?

```python
from gcloud.pubsub.client import Client
from gcloud.pubsub.connection import Connection
connection = Connection.from_service_account_json('/path/to/creds.json')
client = Client(connection=connection)
```

?

Why couldn't we make the code look like...

```python
from gcloud import pubsub
client = pubsub.Client(project_id='my-project', keyFile='/path/to/creds.json')
```

?

Why would we force anyone to know about a "Connection"? (To be honest, I don't even know what a connection is anymore.. when we started, Connection was akin to a Boto connection, which is similar to what I'm now having to call a Client...)